File: src/main/java/io/junseok/mealmate/domain/memberwish/dto/response/MemberWishList.java
package io.junseok.mealmate.domain.memberwish.dto.response;

import io.junseok.mealmate.domain.memberwish.entity.MemberWish;
import io.junseok.mealmate.domain.restaurant.dto.response.RestaurantInfo;
import java.util.LinkedList;
import java.util.List;
import lombok.Builder;

@Builder
public record MemberWishList(List<RestaurantInfo> restaurantInfoList) {
    public static RestaurantInfo fromRestaurantInfo(MemberWish memberWish){
        return RestaurantInfo.builder()
            .restaurantId(memberWish.getRestaurant().getRestaurantId())
            .restaurantName(memberWish.getRestaurant().getRestaurantName())
            .restaurantImageUrl(memberWish.getRestaurant().getRestaurantImageUrl())
            .restaurantType(memberWish.getRestaurant().getRestaurantType())
            .likeCount(memberWish.getRestaurant().getLikeCount())
            .build();
    }
}

File: src/main/java/io/junseok/mealmate/domain/memberwish/dto/response/MemberWishList.kt
package io.junseok.mealmate.domain.memberwish.dto.response;

import io.junseok.mealmate.domain.memberwish.entity.MemberWish
import io.junseok.mealmate.domain.restaurant.dto.response.RestaurantInfo


data class MemberWishList(val restaurantInfoList: List<RestaurantInfo>)

fun MemberWish.toCreateRestaurantInfo() = RestaurantInfo(
    restaurantId = this.restaurant.restaurantId!!,
    restaurantName = this.restaurant.restaurantName,
    restaurantImageUrl = this.restaurant.restaurantImageUrl,
    restaurantType = this.restaurant.restaurantType,
    likeCount = this.restaurant.likeCount,
    openAt = this.restaurant.openAt,
    closeAt = this.restaurant.closeAt
)
